### ISSUE TITLE
Improve Documentation: Synchronized Batch Normalization

### ISSUE THREAD

**Issue by marcvanzee** [google/flax#850]
`axis_index_groups` can be used to set for vanilla batchnorm (aggregate from a single device) or synchronized batchnorm (aggregate from all devices). It controls how large a device group you want to sync over. The current example in the comment is a bit confusing.

If you just want to aggregate over all devices in the axis you don't need to pass axis_index_groups though

**Comment by marcvanzee** [comment]
Closing since I've added this content to the discussion post of #921
